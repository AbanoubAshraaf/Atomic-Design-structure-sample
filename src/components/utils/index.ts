export const isValidAge = (age:number): boolean => {
    return (
      age > 0 && // Age must be positive
      !isNaN(age) && // Age must be a valid number
      Number.isInteger(age) // Age must be an integer
    );
  };
  