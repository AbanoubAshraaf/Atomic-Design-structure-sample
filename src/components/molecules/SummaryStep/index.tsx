import React from "react";
import Link from "next/link";
import { SummaryStepProps } from "@/components/types";

const SummaryStep: React.FC<SummaryStepProps> = (props) => {
  const {
    collectedData: { age, email },
  } = props;
  return (
    <div className="flex flex-col gap-4 w-full">
      <div className="flex flex-col gap-2 items-start">
        <div>Email: {email}</div>
        <div>Age: {age}</div>
      </div>
      <Link className="btn btn-primary" href="/purchased=dev_ins">
        Purchase
      </Link>
    </div>
  );
};

export default SummaryStep;
