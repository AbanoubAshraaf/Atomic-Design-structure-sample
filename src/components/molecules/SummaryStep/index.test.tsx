import React from "react";
import { render, screen } from "@testing-library/react";
import SummaryStep from "@/components/molecules/SummaryStep";
import {
  mockValidDesignerUserData,
  mockValidDeveloperUserData,
} from "@/components/test/mocks";
import { createSnapShot } from "@/components/test/utils";

describe("<SummaryStep/> Component tests", () => {
  test("displays collected data and Purchase button correctly for developer insurance", () => {
    const { container } = render(
      <SummaryStep collectedData={mockValidDeveloperUserData} />,
    );

    expect(screen.getByText("Email: test@getsafe.com")).toBeInTheDocument();
    expect(screen.getByText("Age: 20")).toBeInTheDocument();
    expect(screen.getByText("Purchase")).toBeInTheDocument();
    createSnapShot(container);
  });

  test("displays collected data and Purchase button correctly for designer insurance", () => {
    const { container } = render(
      <SummaryStep collectedData={mockValidDesignerUserData} />,
    );

    expect(screen.getByText("Email: test@getsafe.com")).toBeInTheDocument();
    expect(screen.getByText("Age: 20")).toBeInTheDocument();
    expect(screen.getByText("First Name: John")).toBeInTheDocument();
    expect(screen.getByText("Last Name: Doe")).toBeInTheDocument();
    expect(screen.getByText("Purchase")).toBeInTheDocument();
    createSnapShot(container);
  });
});
