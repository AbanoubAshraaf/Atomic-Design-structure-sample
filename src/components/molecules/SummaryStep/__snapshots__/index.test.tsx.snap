// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SummaryStep/> Component tests should match snapshot 1`] = `
<div>
  <div
    class="flex flex-col gap-4 w-full"
  >
    <div
      class="flex flex-col gap-2 items-start"
    >
      <div>
        Email: 
        test@getsafe.com
      </div>
      <div>
        Age: 
        20
      </div>
    </div>
    <a
      class="btn btn-primary"
      href="/purchased=dev_ins"
    >
      Purchase
    </a>
  </div>
</div>
`;
